src/Controller/ApplePay.php

```diff
@@ -134,4 +137,103 @@ public function get_icon()
 		return $icons_str;
 	}
 
+	/**
+	 * @param \WC_Order $order
+	 * @param int $amount
+	 * @param int $customer_id
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	public function process_standard_payment($order, $amount, $customer_id)
+	{
+		$order_id = PayplugWoocommerceHelper::is_pre_30() ? $order->id : $order->get_id();
+		try {
+			if (!$this->checkApplePay()) {
+				throw new \Exception(__('Payment processing failed. Please retry.', 'payplug'));
+			}
+
+			$address_data = PayplugAddressData::from_order($order);
+
+			$return_url = esc_url_raw($order->get_checkout_order_received_url());
+
+			if (!(str_starts_with($return_url, "http"))) {
+				$return_url = get_site_url().$return_url;
+			}
+
+			$payment_data = [
+				'amount'           => $amount,
+				'currency'         => get_woocommerce_currency(),
+				'payment_method' => 'apple_pay',
```

Comment:
shoudn't this be `$this->id`?

Reply:
for support reasons :)

Reply:
yes I forgot this... it should be $this->id